plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineProjectDeployerTest.java

```diff
@@ -0,0 +1,36 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import static org.mockito.Mockito.when;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AppEngineProjectDeployerTest {
+
+  @Mock IPath stagingDirectory;
+  @Mock CloudSdk cloudSdk;
+  @Mock IProgressMonitor monitor;
+  @Mock AppEngineDeployInfo deployInfo;
+
+  @Test(expected = CoreException.class)
+  public void testDeploy_NoAppEngineProjectId() throws CoreException {
```

Comment:
Shouldn't we be able to deploy a module/service by explicitly specifying a project ID on the command-line (so to speak)?  So the module/service could be deployed to multiple projects?